bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/BagUnpackerWorker.scala

```diff
@@ -40,15 +52,45 @@ case class BagUnpackerWorker(alpakkaSQSWorkerConfig: AlpakkaSQSWorkerConfig,
         val location =
           BagLocationBuilder.build(unpackBagRequest, bagUnpackerWorkerConfig)
         for {
-          unpackSummary <- unpacker.unpack(
+          summaryResult <- unpacker.unpack(
             unpackBagRequest.requestId.toString,
             unpackBagRequest.sourceLocation,
             location.objectLocation)
-          _ <- ingestUpdater.send(unpackBagRequest.requestId, unpackSummary)
+          stepResult = stepResultFor(summaryResult)
+          _ <- ingestUpdater.send(unpackBagRequest.requestId, stepResult)
           _ <- outgoingPublisher.sendIfSuccessful(
-            unpackSummary,
+            stepResult,
             BagRequest(unpackBagRequest.requestId, location))
-        } yield toResult(unpackSummary)
+        } yield toResult(stepResult)
+    }
+
+  private def stepResultFor(result: WorkerResult[UnpackSummary]) = {
+    result match {
+      case workSucceeded: WorkerSucceeded[UnpackSummary] =>
+        IngestStepSucceeded(workSucceeded.summary)
+      case WorkerFailed(
+          unpackSummary: UnpackSummary,
+          archiveLocationException: ArchiveLocationException) =>
+        IngestFailed(
+          unpackSummary,
+          archiveLocationException,
+          Some(clientMessageFor(archiveLocationException)))
+      case WorkerFailed(s: UnpackSummary, e: Throwable) =>
+        IngestFailed(s, e)
     }
+  }
+
+  private def clientMessageFor(exception: ArchiveLocationException) = {
+    val cause = exception.getCause.asInstanceOf[AmazonS3Exception]
+    val archiveLocation = exception.getObjectLocation
+    cause.getStatusCode match {
+      case 403 => s"access to $archiveLocation is denied"
+      case 400 => s"$archiveLocation is invalid"
```

Comment:
I might drop the 400 error code, because “invalid” isn’t a so helpful message, and there might be other S3 exceptions that drop this code? Others look fine.